sources/api/schnauzer/src/helpers.rs

```diff
@@ -2737,4 +2737,151 @@ mod test_oci_spec {
             r#"{ "type": "RLIMIT_NOFILE", "hard": 1, "soft": 2 }"#
         );
     }
+
+    #[test]
+    fn oci_spec_max_address_space_resource_limits_test() {
+        let json = json!({"max-address-space": {"hard-limit": 3, "soft-limit": 4}});
+        let rendered = oci_spec_resource_limits(&json).unwrap();
+        assert_eq!(rendered, r#"{ "type": "RLIMIT_AS", "hard": 3, "soft": 4 }"#);
+    }
```

Comment:
I'd like to see a test to confirm `unlimited` makes it through since we do some custom deserialization and e2e tests are a nice way to make sure things haven't shifted under us when making changes.